### ISSUE TITLE
[Addon: A11y + Addon: Backgrounds] - Results don't change when switching the background color and rerunning a11y

### ISSUE THREAD

**Issue by CodeByAlex** [storybooks/storybook#6000]
**Describe the bug**
When switching the background color of the iframe using the background addon, the results do not change even in the worst case scenario for the a11y addon.

**To Reproduce**
1. Set up project
2. Add a11y and backgrounds
3. render component and switch background color to a color that would fail a11y
4. Re-run accessibility tests
5. Notice that tests results do not change

**Expected behavior**
Expect the a11y addon to recognize changes in background color (ex. Contrast not sufficient)

**Screenshots**
![background-change-a11y-issue](https://user-images.githubusercontent.com/29763590/54085998-c5e76e00-431a-11e9-95e4-1dcb45daf427.gif)

**System:**
 - Browser: [chrome, firefox]
 - Framework: [Angular]  (but I dont believe this is Angular specific)
 - Version: [5.1.0-alpha.3]

**Comment by shilman** [comment]
Hmm, I'm not sure this is actually a bug. In my opinion, the backgrounds tool is a display helper and the a11y tool is testing the component in isolation.

At any rate, given what I understand of the a11y tool and how things are set up it would be quite challenging to get this working without major changes.  Closing this as "won't fix" but anybody who wants to is welcome to disagree, or better yet take a cut at fixing it. 😄

**Comment by benediktvaldez** [comment]
I ran into a similar problem today, in my case I was trying out the color blindness emulation and realised that the background isn't affected by it. IMHO that sort of negates the purpose of the emulation.

My initial reaction @shilman is to disagree. You're right that the backgrounds tool is a display helper, but accessibility is in so many ways exactly about the visual aspect, which leads me to the conclusion that this addon taking the background addon into account is crucial to it's usefulness.

In our specific use case my hope was that the QA team could use the storybook UI to quickly and easily test a lot of those minor issues like color contrasts. Amongst other things that involves trying out different combos of background color and text color since we have a pretty dynamic set of re-usable components. Obviously this could be solved with the knobs addon, but using the background addon is the only way I found to use the grid in combination with a dynamic background.

What do you think @shilman ? cc codeowners of addons/a11y @jbovenschen and @CodeByAlex

**Comment by shilman** [comment]
@benediktvaldez if you can make it happen, go for it!

**Comment by zchsh** [comment]
I ran into this problem as well! I also noticed that the `backgrounds` add-on doesn't show the default background when you click "Open canvas in new tab" button.

My workaround has been to add a `<style>` tag within the story that sets the background colour I want on `body`. Since it's part of the story itself, that `body` style seems to be applied within the story's `iframe`, so the "Open canvas in a new tab" shows the correct background. Of course this no longer uses the background add-on, and requires a knob to adjust the background colour within that `<style>` tag... But it produces what I find to be the most useful behaviour for both the "Open canvas in a new tab" and for the accessibility add-on.

Maybe it'd be possible to have the background add-on set the `background-color` on the story's iframe's `body`, rather than on the `#storybook-preview-background` element outside the iframe as it seems to do now? Or maybe that's not possible, or has other effects I'm missing?
